matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/crypto/MXCryptoConfig.kt

```diff
@@ -31,5 +31,11 @@ data class MXCryptoConfig constructor(
          * If set to false, the request will be forwarded to the application layer; in this
          * case the application can decide to prompt the user.
          */
-        val discardRoomKeyRequestsFromUntrustedDevices: Boolean = true
+        val discardRoomKeyRequestsFromUntrustedDevices: Boolean = true,
+
+        /**
+         * Currently megolm keys are requested to the sender device and to all of our devices.
+         * You can limit request only to your sessions by turning this setting to `true`
+         */
+        val limitRoomKeyRequestsToMyDevices: Boolean = false
```

Comment:
Does this PR replaces #4333 ? If yes, can you close #4333 with a quick comment please?

Reply:
Also I understand that this setting is not dynamic (i.e. cannot be changed by the user). Is it OK?

Reply:
Good point will close 4333, I have done it to adress this point precisly. It's not user configurable, don't think there is a request for it.
I also changed the `requestKeysOnFail`, previously was true only if cross signing was enabled and forks had to modify that code to set it to true. Now we have a mechanism that detects if cross signing is enabled and it will retry to send key request properly (as trust change, outcome of the request might change so worth retrying)